File: src/models/request/report.request.ts
import {IsIn, Length, Matches} from "class-validator"
import { RequestBody } from "../request.js"
import { idRegex } from "./objectId.request.js"

const types = [
	"edit", "report", "other"
] as const;

export class Report extends RequestBody {
	@Matches(idRegex)
	id !: string;
	
	@IsIn(types)
	type !: typeof types[number];
	
	@Length(10, 1200)
	message !: string;
}

File: Models/Request/ReportRequest.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using MongoDB.Bson;

namespace transdb_backend_net.Models.Request;

public enum ReportType
{
    Report,
    Other,
}

public class ReportRequest
{
    [Required(ErrorMessage = "required")]
    public ObjectId Id { get; set; }

    [Required(ErrorMessage = "required")]
    [EnumDataType(typeof(ReportType), ErrorMessage = "invalid_value")]
    public ReportType Type { get; set; }

    [StringLength(2000, ErrorMessage = "length")]
    [Required(ErrorMessage = "required")]
    public string Message { get; set; }
}
